### PR TITLE
fix: Python3.11 (str, Enum) issue

### PR DESCRIPTION
**Pull request by EugeneTorap**
### SUMMARY
Changed Enum.__format__() (the default for format(), str.format() and f-strings) to always produce the same result as Enum.__str__(): for enums inheriting from ReprEnum it will be the member's value; for all other enums it will be the enum and member name (e.g. Color.RED).

Fix the next Py3.11 error
```bash
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "CtasMethod"
LINE 1: DROP CtasMethod.TABLE IF EXISTS test_sync_table
```

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: .github/workflows/superset-python-integrationtest.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        python-version: ["3.9", "3.10"]
+        python-version: ["3.11"]
```
Comment (john-bodley) [inline .github/workflows/superset-python-integrationtest.yml:81]: I think we still want this to be running on the lowest version of Python we support.

Comment (EugeneTorap) [reply .github/workflows/superset-python-integrationtest.yml:81]: It's temporal Py3.11 CI for testing purposes.
I will revert the first commit later.

**Comment by mdeshmu** [comment]
Please look into 3 failing tests.

**Comment by EugeneTorap** [comment]
`Python-Integration / test-postgres (3.11)` CI is passed with the changes

File: superset/utils/backports.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (john-bodley) [inline superset/utils/backports.py:1]: Shouldn't we be using https://pypi.org/project/backports.strenum/ or something similar?

File: .pylintrc [hunk 3]
```diff
@@ -75,6 +75,7 @@ enable=
 # --disable=W"
 disable=
     cyclic-import,  # re-enable once this no longer raises false positives
+    no-member,  # re-enable once this no longer raises false positives
```
Comment (john-bodley) [inline .pylintrc:78]: I mentioned the my concerns about blanket disabling of messages in the past. Is this a necessary requirement?

File: superset/utils/backports.py (hunk 2 above)
Comment (EugeneTorap) [reply superset/utils/backports.py:1]: No, I tested `backports.strenum` and there are mypy issues

File: .pylintrc (hunk 3 above)
Comment (EugeneTorap) [reply .pylintrc:78]: Yes, it's necessary requirement before we update min supported python to 3.11

Comment (villebro) [reply .pylintrc:78]: Maybe we could add a note in the comment that this will become redundant after the min required version is 3.11

**Review by villebro — APPROVED** [review]
LGTM, would love @john-bodley 's approval for merging

File: .pylintrc (hunk 3 above)
Comment (mdeshmu) [reply .pylintrc:78]: meanwhile raised the issue on pylint GitHub https://github.com/pylint-dev/pylint/issues/8897

**Comment by mdeshmu** [comment]
I think, we also need to update files where [`str, enum.Enum`](https://github.com/search?q=repo%3Aapache%2Fsuperset%20%22str%2C%20enum.Enum%22&type=code) is used.

**Comment by EugeneTorap** [comment]
Thanks @mdeshmu! Updated it.

File: .pylintrc (hunk 3 above)
Comment (EugeneTorap) [reply .pylintrc]: @villebro Added the note in the comment

**Review by john-bodley — APPROVED** [review]

**Pull request merged**